pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -389,7 +389,10 @@ class Timestamp(_Timestamp):
             # User passed tzinfo instead of tz; avoid silently ignoring
             tz, tzinfo = tzinfo, None
 
-        if isinstance(ts_input, str):
+        if isinstance(ts_input, Timestamp) and tz is None:
```

Comment:
you're going to need to check that all the other kwargs are None

Reply:
Thanks, I'll change this, but first I need to solve #24329, because right now we rely on Timestamp constructor changing ts.value when called on a localized DST Timestamp (this is why the tests for this PR break currently).

Reply:
Depending on how digging through the code goes, this PR might also take care of #24329, but I still need a couple days to analyze tz_localize, pd.date_range, and the Timestamp constructor.

Reply:
@jbrockmendel Solved #24329. Now this PR can be reviewed again.